src/main/java/com/rummgp/medical_clinic/service/PatientService.java

```diff
@@ -0,0 +1,49 @@
+package com.rummgp.medical_clinic.service;
+
+import com.rummgp.medical_clinic.model.Patient;
+import com.rummgp.medical_clinic.repository.PatientRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class PatientService {
+
+    private final PatientRepository patientRepository;
+
+    public List<Patient> getAll() {
+        return patientRepository.findAll();
+    }
+
+    public Patient getPatient(String email) {
+        return patientRepository.findByEmail(email)
+                .orElseThrow(() -> new IllegalArgumentException("Nie ma takiego pacjenta"));
+    }
+
+    public void addPatient(Patient patient) {
```

Comment:
if (!patientRepository.add(patient)) {
          throw new EmailAlreadyExistsException("Email " + patient.getEmail() + " is already taken");
}
return true;